server/src/main/java/org/jetbrains/bsp/bazel/server/sync/ModuleMerger.java

```diff
@@ -0,0 +1,126 @@
+package org.jetbrains.bsp.bazel.server.sync;
+
+import io.vavr.collection.HashSet;
+import io.vavr.collection.Seq;
+import io.vavr.control.Option;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.bsp.bazel.server.sync.languages.LanguageData;
+import org.jetbrains.bsp.bazel.server.sync.languages.java.JavaModule;
+import org.jetbrains.bsp.bazel.server.sync.languages.scala.ScalaModule;
+import org.jetbrains.bsp.bazel.server.sync.model.Label;
+import org.jetbrains.bsp.bazel.server.sync.model.Module;
+import org.jetbrains.bsp.bazel.server.sync.model.SourceSet;
+
+public class ModuleMerger {
```

Comment:
Why do we need this? IntelliJ gets incompatible modules from bsp and merges them on its own.

Reply:
+ in the future the client will handle such case even better

Reply:
I've had a few people say this now (that IJ handles merging modules), in practice I've rarely seen it actually work correctly.  Can you point me to where it does this?

Reply:
it's inside current bsp client (https://github.com/JetBrains/intellij-scala/tree/idea221.x/bsp) - it creates a single IJ module using multiple bsp modules - it just merges all the sources / opts / names and creates single IJ module. We are slowly working on a new client, with better support for that

Reply:
bump i believe that we dont want to do it here - it's kinda agains protocol, it might have strange conflicts with the clients mechanism + new IJ BSP client solves it

Reply:
We should make projectviews more flexible to put there and read config easily. Perhaps add a generic feature flags (so that we can do for example `bazel-bsp -t //my/target/... -f merge-modules=true -f add-synthetic-module-roots=true`. Such feature would be opt in. If it really makes bazel-bsp work better in intellij right now, I would take it. There is similar situation with my synthetic modules. They should be behind a flag. What should be the default is up for discussion.